lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -51,6 +52,11 @@
             }
         }
 
+        // Splitting IndexOf in several smaller methods make it passing the requirements for inlining.
```

Comment:
>Splitting IndexOf in several smaller methods make it passing the requirements for inlining [](start = 11, length = 90)

That's probably fine for now, but we should instead tweak the inlining heuristics.
Since we know this is a builtin function, we should be able to bypass the inlining checks